tomcat-native.yaml

```diff
@@ -0,0 +1,67 @@
+package:
+  name: tomcat-native
+  version: 2.0.4
+  epoch: 0
+  description: OpenSSL extension for Tomcat
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - ca-certificates-bundle
+      - build-base
+      - busybox
+      - openjdk-17
+      - libapr
+      - libapr-dev
+      - ant
+      - openssl
+      - openssl-dev
+      - autoconf
+      - python3
+  environment:
+    JAVA_HOME: /usr/lib/jvm/java-17-openjdk
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/tomcat-native
+      tag: ${{package.version}}
+      expected-commit: 2f080011a624527420c202331c0eb4880c7cc035
+
+  - runs: |
+      ant jar
+
+      mkdir -p ${{targets.destdir}}/usr/share/tomcat/lib
+      cp dist/tomcat-native-${{package.version}}.jar ${{targets.destdir}}/usr/share/tomcat/lib
+
+  - runs: |
+      cd native
+      ./buildconf --with-apr=/usr/share/apr-1
+      ./configure --prefix=/usr --with-java-home=$JAVA_HOME --with-ssl=/usr/include/openssl --with-apr=/usr/bin/apr-1-config
+
+  - uses: autoconf/configure
+    with:
+      dir: native
+      opts: |
+        --prefix=/usr \
+        --with-java-home=$JAVA_HOME \
```

Comment:
tomcat-native needs the CATALINA_HOME

Reply:
I think `CATALINA_HOME` home here is `/usr/share/tomcat`. The jar for tomcat-native gets installed to `/usr/share/tomcat/lib` and the C libs to `/usr/lib`.

https://tomcat.apache.org/native-doc/#Building/UNIX

Looks like Arch does the same thing when installed as a system package https://gitlab.archlinux.org/archlinux/packaging/packages/tomcat-native/-/blob/main/PKGBUILD#L48

Reply:
I can confirm this works with loading up tomcat-native as is.

Not sure what's right idiomatically though. Maybe @rawlingsj does.

```
 chainguard/wolfi-os[ez/add-tomcat ☂ ] docker run --rm -i -t tomcat:local-amd64
Using CATALINA_BASE:   /usr/share/tomcat
Using CATALINA_HOME:   /usr/share/tomcat
Using CATALINA_TMPDIR: /usr/share/tomcat/temp
Using JRE_HOME:        /usr/lib/jvm/java-17-openjdk
Using CLASSPATH:       /usr/share/tomcat/bin/bootstrap.jar:/usr/share/tomcat/bin/tomcat-juli.jar
Using CATALINA_OPTS:
07-Jun-2023 22:28:04.043 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version name:   Apache Tomcat/10.1.10
07-Jun-2023 22:28:04.046 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Jan 1 1970 00:00:00 UTC
07-Jun-2023 22:28:04.046 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version number: 10.1.10.0
07-Jun-2023 22:28:04.046 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            6.3.6-arch1-1
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/lib/jvm/java-17-openjdk
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           17.0.8-internal+0-wolfi-r0
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            wolfi
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/share/tomcat
07-Jun-2023 22:28:04.047 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/share/tomcat
07-Jun-2023 22:28:04.058 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/usr/share/tomcat/conf/logging.properties
07-Jun-2023 22:28:04.059 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
07-Jun-2023 22:28:04.059 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
07-Jun-2023 22:28:04.059 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
07-Jun-2023 22:28:04.059 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
07-Jun-2023 22:28:04.059 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED
07-Jun-2023 22:28:04.059 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED
07-Jun-2023 22:28:04.060 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.util=ALL-UNNAMED
07-Jun-2023 22:28:04.060 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
07-Jun-2023 22:28:04.060 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
07-Jun-2023 22:28:04.060 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/usr/share/tomcat
07-Jun-2023 22:28:04.060 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/usr/share/tomcat
07-Jun-2023 22:28:04.060 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/usr/share/tomcat/temp
07-Jun-2023 22:28:04.064 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent Loaded Apache Tomcat Native library [2.0.4] using APR version [1.7.4].
07-Jun-2023 22:28:04.067 INFO [main] org.apache.catalina.core.AprLifecycleListener.initializeSSL OpenSSL successfully initialized [OpenSSL 3.1.1 30 May 2023]
07-Jun-2023 22:28:04.250 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8080"]
07-Jun-2023 22:28:04.270 INFO [main] org.apache.catalina.startup.Catalina.load Server initialization in [359] milliseconds
07-Jun-2023 22:28:04.318 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
07-Jun-2023 22:28:04.318 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet engine: [Apache Tomcat/10.1.10]
07-Jun-2023 22:28:04.330 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
07-Jun-2023 22:28:04.342 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [71] milliseconds
```

Reply:
Nice !!!